### PR TITLE
feat: set namecards in profile editing.

### PR DESCRIPTION
**Pull request by hamusuke0323**
## Description

Please carefully read the [Contributing note](https://github.com/Grasscutters/Grasscutter/blob/stable/CONTRIBUTING.md) and [Code of conduct](https://github.com/Grasscutters/Grasscutter/blob/development/CODE_OF_CONDUCT.md) before making any pull requests.
And, **Do not make a pull request to merge into stable unless it is a hotfix. Use the development branch instead.**
## Type of changes

- [ ] Bug fix
- [x] New feature 
- [ ] Enhancement
- [ ] Documentation

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] My pull request is unique and no other pull requests have been opened for these changes
- [x] I have read the [Contributing note](https://github.com/Grasscutters/Grasscutter/blob/stable/CONTRIBUTING.md) and [Code of conduct](https://github.com/Grasscutters/Grasscutter/blob/development/CODE_OF_CONDUCT.md)
- [x] I am responsible for any copyright issues with my code if it occurs in the future.

### CODE REVIEW

**Comment by hamusuke0323** [comment]
Need screenshots of when this feature works?

File: src/main/java/emu/grasscutter/game/player/Player.java [hunk 1]
```diff
@@ -968,6 +970,18 @@ public List<ShowAvatarInfoOuterClass.ShowAvatarInfo> getShowAvatarInfoList() {
         return showAvatarInfoList;
     }
 
+    public List<Integer> getShowNameCardInfoList() {
+        Player player;
+        if (this.isOnline()) {
+            player = this;
+        } else {
+            player = DatabaseHelper.getPlayerByUid(id);
+        }
```
Comment (Birdulon) [inline src/main/java/emu/grasscutter/game/player/Player.java]: This is pretty much a reimplementation of the existing pattern:

```suggestion
        Player player = Grasscutter.getGameServer().getPlayerByUid(id, true);
```

Though as a method of `Player`, I'd expect that it should already be an offline player object if it ever gets to the point of fetching the offline player object, so it probably doesn't make sense to do the lookup at all.

File: src/main/java/emu/grasscutter/game/player/Player.java [hunk 2]
```diff
@@ -968,6 +970,13 @@ public List<ShowAvatarInfoOuterClass.ShowAvatarInfo> getShowAvatarInfoList() {
         return showAvatarInfoList;
     }
 
+    public List<Integer> getShowNameCardInfoList() {
+        Player player = Grasscutter.getGameServer().getPlayerByUid(id, true);
+
+        List<Integer> info = player.getShowNameCardList();
+        return info == null ? new ArrayList<>() : info;
```
Comment (Birdulon) [inline src/main/java/emu/grasscutter/game/player/Player.java]:
```suggestion
        List<Integer> info = this.getShowNameCardList();
        return info == null ? new ArrayList<>() : info;
```

File: src/main/java/emu/grasscutter/server/packet/recv/HandlerGetFriendShowNameCardInfoReq.java [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+import emu.grasscutter.net.packet.PacketOpcodes;
+import emu.grasscutter.net.proto.GetFriendShowNameCardInfoReqOuterClass;
+import emu.grasscutter.server.game.GameSession;
+import emu.grasscutter.server.packet.send.PacketGetFriendShowNameCardInfoRsp;
+
+@Opcodes(PacketOpcodes.GetFriendShowNameCardInfoReq)
+public class HandlerGetFriendShowNameCardInfoReq extends PacketHandler {
+    @Override
+    public void handle(GameSession session, byte[] header, byte[] payload) throws Exception {
+        GetFriendShowNameCardInfoReqOuterClass.GetFriendShowNameCardInfoReq req = GetFriendShowNameCardInfoReqOuterClass.GetFriendShowNameCardInfoReq.parseFrom(payload);
```
Comment (Birdulon) [inline src/main/java/emu/grasscutter/server/packet/recv/HandlerGetFriendShowNameCardInfoReq.java]:
```suggestion
        var req = GetFriendShowNameCardInfoReqOuterClass.GetFriendShowNameCardInfoReq.parseFrom(payload);
```

File: src/main/java/emu/grasscutter/server/packet/recv/HandlerUpdatePlayerShowNameCardListReq.java [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+import emu.grasscutter.net.packet.PacketOpcodes;
+import emu.grasscutter.net.proto.UpdatePlayerShowNameCardListReqOuterClass;
+import emu.grasscutter.server.game.GameSession;
+import emu.grasscutter.server.packet.send.PacketUpdatePlayerShowNameCardListRsp;
+
+@Opcodes(PacketOpcodes.UpdatePlayerShowNameCardListReq)
+public class HandlerUpdatePlayerShowNameCardListReq extends PacketHandler {
+    @Override
+    public void handle(GameSession session, byte[] header, byte[] payload) throws Exception {
+        UpdatePlayerShowNameCardListReqOuterClass.UpdatePlayerShowNameCardListReq req = UpdatePlayerShowNameCardListReqOuterClass.UpdatePlayerShowNameCardListReq.parseFrom(payload);
```
Comment (Birdulon) [inline src/main/java/emu/grasscutter/server/packet/recv/HandlerUpdatePlayerShowNameCardListReq.java]:
```suggestion
        var req = UpdatePlayerShowNameCardListReqOuterClass.UpdatePlayerShowNameCardListReq.parseFrom(payload);
```

File: src/main/java/emu/grasscutter/server/packet/send/PacketGetFriendShowNameCardInfoRsp.java [hunk 5]
```diff
@@ -0,0 +1,21 @@
...
+import emu.grasscutter.net.packet.PacketOpcodes;
+import emu.grasscutter.net.proto.GetFriendShowNameCardInfoRspOuterClass;
+
+import java.util.List;
+
+public class PacketGetFriendShowNameCardInfoRsp extends BasePacket {
+    public PacketGetFriendShowNameCardInfoRsp(int uid, List<Integer> cardIds) {
+        super(PacketOpcodes.GetFriendShowNameCardInfoRsp);
+
+        GetFriendShowNameCardInfoRspOuterClass.GetFriendShowNameCardInfoRsp rsp = GetFriendShowNameCardInfoRspOuterClass.GetFriendShowNameCardInfoRsp.newBuilder()
```
Comment (Birdulon) [inline src/main/java/emu/grasscutter/server/packet/send/PacketGetFriendShowNameCardInfoRsp.java]:
```suggestion
        var rsp = GetFriendShowNameCardInfoRspOuterClass.GetFriendShowNameCardInfoRsp.newBuilder()
```

File: src/main/java/emu/grasscutter/server/packet/send/PacketUpdatePlayerShowNameCardListRsp.java [hunk 6]
```diff
@@ -0,0 +1,20 @@
...
+import emu.grasscutter.net.packet.PacketOpcodes;
+import emu.grasscutter.net.proto.UpdatePlayerShowNameCardListRspOuterClass;
+
+import java.util.List;
+
+public class PacketUpdatePlayerShowNameCardListRsp extends BasePacket {
+    public PacketUpdatePlayerShowNameCardListRsp(List<Integer> cardIds) {
+        super(PacketOpcodes.UpdatePlayerShowNameCardListRsp);
+
+        UpdatePlayerShowNameCardListRspOuterClass.UpdatePlayerShowNameCardListRsp rsp = UpdatePlayerShowNameCardListRspOuterClass.UpdatePlayerShowNameCardListRsp.newBuilder()
```
Comment (Birdulon) [inline src/main/java/emu/grasscutter/server/packet/send/PacketUpdatePlayerShowNameCardListRsp.java]:
```suggestion
        var rsp = UpdatePlayerShowNameCardListRspOuterClass.UpdatePlayerShowNameCardListRsp.newBuilder()
```

**Review by Birdulon — CHANGES_REQUESTED** [review]

**Review by Birdulon — APPROVED** [review]

**Pull request merged**